druid/src/widget/scrollbar.rs

```diff
@@ -0,0 +1,375 @@
+use std::mem;
+
+use crate::kurbo::{Rect, RoundedRect, Size};
+use crate::theme;
+use crate::{
+    BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx, RenderContext, Selector,
+    UpdateCtx, Widget,
+};
+
+/// Trait used for shared state between
+/// the scrollbar and the scroll container.
+pub trait ScrollControlState: Data + PartialEq {
```

Comment:
This trait is needed to implement a shared state between the scrollbar and the widget being "scrolled". Normally, this information is derived from the container widget when the scrollbar is implemented as a child. This decoupled scrollbar acts more like a reskinned slider widget, taking in some values to scroll and interpolating between them.

Reply:
I should mention that the `scroll_position` can be influenced from the outside making it possible to scroll to a specific element in the list in response to an action originating anywhere in the app.

Reply:
I'll have to review this pattern and learn how it can be utilized here. Sounds exciting.

Reply:
the best place to start would probably be `examples/identity.rs`. There's a bunch of improvements landing in the next week or so too, 🤞.